py/desitarget/mock/mockmaker.py

```diff
@@ -1054,6 +1054,9 @@ def populate_targets_truth(self, flux, data, meta, objmeta, indx=None,
         if len(objmeta) > 0 and len(objtruth) > 0: # some objects have no metadata...
             for key in objmeta.colnames:
                 if key in objtruth.colnames:
+                    #The modified emision line model don't have the same shape as the default.
+                    if (key=='EMLINES') and (objtruth[key][:].shape != objmeta[key].shape) :
+                        objtruth.replace_column('EMLINES',np.zeros((nobj,len(objmeta['EMLINES'][0, :, 0]), 3))-1)
                     objtruth[key][:] = objmeta[key]
```

Comment:
This code looks pretty fragile, but I don't have another suggestion unfortunately.

Reply:
Yes, I know... but I tried other things and were worse...